pkg/registry/core/service/portallocator/controller/repair.go

```diff
@@ -112,14 +112,17 @@ func (c *Repair) doRunOnce() error {
 	// If etcd server is not running we should wait for some time and fail only then. This is particularly
 	// important when we start apiserver and etcd at the same time.
 	var snapshot *api.RangeAllocation
-
-	err := wait.PollImmediate(time.Second, 10*time.Second, func() (bool, error) {
-		var err error
+	var err error
+	err = wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 30*time.Second, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
```suggestion
	err = wait.PollUntilContextTimeout(context.Background(), time.Second, 30*time.Second, true, func(ctx context.Context) (bool, error) {
```

Reply:
Thank you for the reminder, it's already been removed.